### ISSUE TITLE
Setting the learning rate

### ISSUE THREAD

**Issue by brady990** [ultralytics/ultralytics#13898]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I'm trying to train a yolov8 model that has already been trained with a lot of data by adding a small amount of new data with the same categories as before, but I've noticed that when I set lr0 to a very small value, for example 2e-5, the lr/pg0 of the first epoch of the training will also be very high, while the lr of the following epochs will all be small, resulting in a very low map for the whole training process. I would like to ask how to set the lr/pg0 for the first epoch instead of letting it choose a value automatically?

**Comment by glenn-jocher** [comment]
@brady990 hello,

Thank you for your question and for providing details about your issue.

It sounds like you're experiencing an unexpected behavior with the learning rate during the initial epoch of your training. To ensure we can investigate this thoroughly, could you please provide a minimum reproducible code example? This will help us replicate the issue on our end. You can refer to our guide on creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please make sure you are using the latest versions of `torch` and `ultralytics`. You can update your packages using the following commands:
```bash
pip install --upgrade torch ultralytics
```

Regarding your specific question about setting the learning rate for the first epoch, you can manually adjust the learning rate schedule by modifying the `lr_scheduler` parameters in your training configuration. Here's an example of how you might set this up in Python:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Define a custom learning rate schedule
def custom_lr_scheduler(epoch):
    if epoch == 0:
        return 2e-5  # Set a specific learning rate for the first epoch
    else:
        return 1e-4  # Set a different learning rate for subsequent epochs

# Train the model with the custom learning rate scheduler
model.train(data='coco128.yaml', epochs=100, lr0=2e-5, lr_scheduler=custom_lr_scheduler)
```

This approach allows you to control the learning rate dynamically based on the epoch. If you have any further questions or need additional assistance, please feel free to ask!

**Comment by Y-T-G** [comment]
`model.train(optimizer="SGD", lr0=0.001, warmup_epochs=0)`

**Comment by glenn-jocher** [comment]
Hello @Y-T-G,

Thank you for your question and for providing details about your issue.

It sounds like you're experiencing an unexpected behavior with the learning rate during the initial epoch of your training. To ensure we can investigate this thoroughly, could you please provide a minimum reproducible code example? This will help us replicate the issue on our end. You can refer to our guide on creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please make sure you are using the latest versions of `torch` and `ultralytics`. You can update your packages using the following commands:
```bash
pip install --upgrade torch ultralytics
```

Regarding your specific question about setting the learning rate for the first epoch, you can manually adjust the learning rate schedule by modifying the `lr_scheduler` parameters in your training configuration. Here's an example of how you might set this up in Python:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Define a custom learning rate schedule
def custom_lr_scheduler(epoch):
    if epoch == 0:
        return 2e-5  # Set a specific learning rate for the first epoch
    else:
        return 1e-4  # Set a different learning rate for subsequent epochs

# Train the model with the custom learning rate scheduler
model.train(data='coco128.yaml', epochs=100, lr0=2e-5, lr_scheduler=custom_lr_scheduler)
```

This approach allows you to control the learning rate dynamically based on the epoch. If you have any further questions or need additional assistance, please feel free to ask!
